test/wdio_functional/WDIO_FUNCTIONAL_TESTS.md

```diff
@@ -0,0 +1,269 @@
+# Kibana Test Harness
+
+## Running Tests
+
+### Production Command
+
+`yarn test:ui:wdio`
+
+### To Run Tests Without Restarting Servers
+
+```sh
+# In one terminal start the test server.
+yarn test:ui:server
+```
+```sh
+# In another terminal, run the tests:
+npm run wdioTests
+```
+
+## Writing Tests
+
+### Patterns
+
+#### Page Object Model
+
+##### Navigation
+
+Kibana Test Harness uses the Page Object Model. This basically means that the objects that our tests interact are constructed as pages. When using these in the tests, we interact with instances of those objects. The navigation elements return instances of their corresponding Page Object. 
+
+```javascript
+this.driver.url('/');
```

Comment:
Based on this, it feels like `nav` should be a service rather than a property of page objects, then it could be used to get the first page object rather than needing to create it manually:

```js
const nav = getService('nav');
this.homePage = nav.navigateToHomePage();
this.console = nav.navigateToConsole();
```